napari/components/viewer_model.py

```diff
@@ -39,47 +44,32 @@ class ViewerModel(AddLayersMixin, KeymapHandler, KeymapProvider):
         Preset color palettes.
     """
 
-    themes = palettes
-
-    def __init__(
-        self, title='napari', ndisplay=2, order=None, axis_labels=None
-    ):
-        super().__init__()
-
-        self.events = EmitterGroup(
-            source=self,
-            auto_connect=True,
-            status=Event,
-            help=Event,
-            title=Event,
-            interactive=Event,
-            cursor=Event,
-            reset_view=Event,
-            active_layer=Event,
-            palette=Event,
-            grid=Event,
-            layers_change=Event,
-        )
-
+    title: str = 'napari'
+    status: str = 'Ready'
+    help: str = ''
+    cursor: str = 'standard'
+    cursor_size: Optional[int] = None
+    interactive: bool = True
+    active_layer: Optional[int] = None
+    grid_size: Tuple[int, int] = (1, 1)
+    theme: str = DEFAULT_THEME
+
+    grid_stride: ClassVar[int] = 1
+    palette: ClassVar[Dict[str, str]] = palettes[DEFAULT_THEME]
+    layers: ClassVar[LayerList] = []
+
+    ndisplay: InitVar[int] = 2  # type: ignore
+    order: InitVar[Optional[Tuple[int, ...]]] = None  # type: ignore
+    axis_labels: InitVar[Optional[Sequence[str]]] = None  # type: ignore
```

Comment:
Interesting with the `InitVar` (see https://docs.python.org/3/library/dataclasses.html#post-init-processing) for those unfamiliar with it.

Will `InitVar` not get their own event emission? Can they be modified later? I guess not - see
`They are not otherwise used by dataclasses.` in the docs. I think that's ok though, just want to check.

Reply:
Yeah, that's by design.  These are variables that are required to initialize the dataclass state, but have no independent meaning after that.  So there are no events (because they don't exist after initialization).
Think of them as parameters that get used only during init (which is how those three are used here)
```python
class Test:
    def __init__(self, a, b):
        self.c = a * b
    # a and b are gone now...
```
if they are actually important attributes for the dataclass, just remove the InitVar class and they will have events.

and, as you probably saw in the docs, `InitVars` will be passed to the `__post_init__` method.